### ISSUE TITLE
[3.3.5] Pet Talent Cobra Reflexes Should Not Play Animation

### ISSUE THREAD

**Issue by MrSmite** [TrinityCore/TrinityCore#19294]
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

**Description:**

[Cobra Reflexes](http://wotlk.openwow.com/spell=25077) is a passive Hunter pet talent that currently incorrectly has an audio and visual component when the pet is added to the world.

**Current behaviour:**

See description

**Expected behaviour:**

Because Cobra Reflexes is a passive talent it should not have an audio / visual display when the pet is added to the world, similar to other passive talents such as [Spiked Collar](http://wotlk.openwow.com/spell=53182). Only proc talents such as [Frenzy](http://wotlk.openwow.com/spell=19624) trigger audio and visual components.

NOTE: I could not find any specifics on wowhead about this spell because of Legion and there isn't really any good "evidence" on YouTube. The basis of my report is simply that no other passive talents have an audio and visual component applied.

**Steps to reproduce the problem:**

1. Select Cobra Reflexes in the Hunter pet talent tree
2. Do something such as zoning, dismounting or call pet to trigger adding the pet to the world
3. Notice the giant cobra animation and hissing sound that is incorrectly displayed

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

https://github.com/TrinityCore/TrinityCore/commit/8c6fa411e8118752ddbefe8dc022514602859fb9

> *CHANGEME Copy the first line of the `worldserver`*
> Question: On Windows, how do you copy text out of the console? Right-click does nothing and CTRL+C stops the server...

**TDB version:**

TDB_full_world_335.61_2016_04_11 + updates

**Operating system:**

WinXP / Win7

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by mario-64** [comment]
Thank you for reporting this issue. It's existed for a long time. You're correct passive talents should make no animation or sound when a pet is called

**Comment by MrSmite** [comment]
> It's existed for a long time.

Yep, I've been meaning to look into it myself but haven't had the time.

**Comment by ariel-** [comment]
Should be added to pet using AddAura instead of CastSpell then.

> Question: On Windows, how do you copy text out of the console? Right-click does nothing and CTRL+C stops the server...

Right click on the console window, select Mark (or use default shortcut Ctrl-M), highlight your text, then press Enter and it will be copied to clipboard

**Comment by MrSmite** [comment]
> Right click on the console window, select Mark (or use default shortcut Ctrl-M)

Yeah, tried that already because I assumed it was a standard console window. There is no menu so I can't select "Mark" Also, CTRL+M only prints "TC>" in the console but still does not let me mark anything.

**Comment by ariel-** [comment]
Right click on title bar, and go to the "Edit" submenu then

**Comment by MrSmite** [comment]
> Right click on title bar, and go to the "Edit" submenu then

Ah yes, that worked. I get so used to using shortcuts I completely forgot about that. Thanks for reminding me.

**Comment by MrSmite** [comment]
I changed calls for *CastSpell()* to *AddAura()* and it works. I also tested other talents (player and pet) and abilities (Cower / Furious Howl) and those seem to work correctly too.

Do all spells / talents work with *AddAura()* or is there a chance something will actually need *CastSpell()* to apply properly?

__Pet::addSpell()__

```C++
    if (spellInfo->IsPassive() && (!spellInfo->CasterAuraState || HasAuraState(AuraStateType(spellInfo->CasterAuraState))))
        AddAura(spellId, this);
        //CastSpell(this, spellId, true);
```

__Pet::CastPetAura()__

```C++
    else
        AddAura(auraId, this);
        //CastSpell(this, auraId, true);
```

**Comment by mario-64** [comment]
Can this be merged now?

**Comment by MrSmite** [comment]
> Can this be merged now?

I honestly don't know. What I posted above works and doesn't seem to interfere with other talents but I only tested a few and that's assuming the other talents aren't broken themselves without this fix.

**Comment by CraftedRO** [comment]
after watching even more videos (https://youtu.be/bg2YPh3afyM?t=246 https://youtu.be/y6h9Dp7xTsI?list=PLf_ZOy0QRT_C2CTKRgdKc1_PLOWx_SfDI&t=351 / + https://youtu.be/y6h9Dp7xTsI?list=PLf_ZOy0QRT_C2CTKRgdKc1_PLOWx_SfDI&t=9237), it turns out the animation only shoud appear when you call it from stable, when you spec into the talent or you call pet or you land from a flight master it shoud not appear so I guess we have to reopen and maybe we have to use add aura as suggested, my bad will try to look into code

**Comment by MrSmite** [comment]
Sorry, I don't have any of my old patches or I would post it.
